osrf/vrx
# [Garden] Port scan, dock, deliver - Part 3

This pull request finishes porting the ScanDockScoringPlugin from Gazebo classic. In particular, it ports the functionality related with the detection of the projectile in the placard targets.

### How to test it?

1. Apply the following patch to move the vehicle to the central bay.

    <details><summary>patch</summary>

    ````diff
    diff --git a/vrx_gz/launch/competition.launch.py b/vrx_gz/launch/competition.launch.py
    index 9d13e862..2590106a 100644
    --- a/vrx_gz/launch/competition.launch.py
    +++ b/vrx_gz/launch/competition.launch.py
    @@ -39,7 +39,7 @@ def launch(context, *args, **kwargs):
             with open(config_file, 'r') as stream:
                 models = Model.FromConfig(stream)
         else:
    -      m = Model('wamv', 'wam-v', [-532, 162, 0, 0, 0, 1])
    +      m = Model('wamv', 'wam-v', [-486.4, 197, 0, 0, 0, 1])
           if robot_urdf and robot_urdf != '':
               m.set_urdf(robot_urdf)
           models.append(m)

* Recompile and launch the scan_dock_deliver_task world:
```
ros2 launch vrx_gz competition.launch.py world:=scan_dock_deliver_task
```

* On a new terminal launch the joystick teleop node:
```
ros2 launch vrx_gz usv_joy_teleop.py
```

* On a new terminal check the `/vrx/task/info` topic:
```
ros2 topic  echo /vrx/task/info
```

* Let's shoot a projectile. Open a new terminal and run:
```
ros2 topic pub --once /wamv/shooters/ball_shooter/fire std_msgs/Empty
```

If everything goes well, the ball should go inside the big target. Verify that the scoring increments in `5` points each time a target is hit. Remember that we only allow `4` attempts.
